### DISCUSSION TITLE
What would it take for the Flathub build to be supported?

### DISCUSSION THREAD

**Discussion by cassidyjames** [audacity/audacity#5892] [Hot potatoes]
Hey all! I'm a volunteer with [Flathub](https://flathub.org), the Linux app store with over a million active users, backed by KDE and GNOME, usable on all modern Linux distros, and included out of the box on Steam Deck desktop mode, Clear Linux, Endless OS, Fedora Linux, KDE Neon, Linux Mint, Pop!_OS, Zorin OS, and more.

As you probably know, a community-maintained build of Audacity is [available on Flathub](https://flathub.org/apps/org.audacityteam.Audacity) with over 500k installs. I was originally reaching out to see if you all would be interested in [verifying your app](https://docs.flathub.org/docs/for-app-authors/verification), but in doing my research I see that the team always directs folks to the Flathub issue tracker and says the Flatpak is not supported.

As someone who uses Audacity on Endless OS and Fedora Silverblue, Flatpak is the only way I install apps—and it's clearly the preferred cross-desktop format that the desktop environments and distros are all adopting and collaborating on. So I was wondering: **what would need to happen for the Flatpak to be supported?** Is it a technical limitation, a social decision, a maintenance concern, or something else? I'm aware I've posted in the **☢️ Hot potatoes** category, so I'm prepared to hear real talk, here. :)

I'm also happy to help out however I can and answer any questions you may have about Flatpak or Flathub!

**Comment by crsib** [comment]
We have quite a few reasons why we don't do Flatpak internally:

1. Flatpak adds another level of complication to the build. We put effort into making Audacity build without using the default dependency management, but we do not have the resources to test the result. For this reason, we leave Audacity packaging to the package maintainers. There are bugs related specifically to using different versions of dependencies. We fix them, but we do not commit to fixing them promptly. 
2. Flatpak adds a level of complication for running as well. 
3. We want the "official" builds to work roughly the same across the platforms, including the audio.com related features. 

AppImage, despite its compromises, allows us to distribute the binaries built with the features we want enabled.

**Reply by cassidyjames** [reply]
Thanks for the detailed reply!

I understand where you're coming from with your first point a bit; though one might argue that the advantage of Flatpak is that you as the developer can define precisely what dependencies are used in the build, including exact releases or commit hashes of third-party projects. If the current community-maintained Flatpak is not matching the versions of dependencies that upstream (you) desire or require, then that should be a pretty easy fix by correcting the Flatpak manifest to point to those correct versions.

For 2, I'm not sure I understand what complexity it adds; are you referring to any specific limitations of sandboxing?

3 sounds like an argument _in favor_ of Flatpak to me, given the above; what is preventing using audio.com related features in a Flatpak?

Is the disconnect that the Flatpak is currently community-maintained, and thus those community maintainers have made different choices than Audacity would make? If that's the case, I think it would be entirely reasonable to correct that and align with your desires—of course that would be easiest if Audacity were maintaining the Flatpak manifest itself, but I think we could still better align under the current setup.

**Reply by crsib** [reply]
1. That is still a complication - we need to have 2 separate dependency management solutions. Audacity is a cross-platform application; Flatpak can't be "one ring to rule them all"
2. We do receive bugs related to sandboxing, and we forward them to the Flatpak maintainers :-)
3. That is a very sensitive and complicated topic. Building Audacity with networking enabled will surely cause a lot of tension in the Linux community. You can see a very light example of that at https://github.com/flathub/org.audacityteam.Audacity/issues/147

**Reply by cassidyjames** [reply]
I appreciate the follow-up and candor. I'm here because I would love to see more apps shipping to Flathub for their official Linux build, simply because it makes my life easier when telling people where to get their apps—and my work with GNOME and Endless OS means I have an incentive to make those ecosystems more app-rich and well-supported by app developers!

1. I wonder if it could be automated to pull in from your upstream dependency management; the Flathub manifest is a pretty straightforward format, and it looks like there's [quite a bit of scripting](https://github.com/audacity/audacity/tree/master/linux) to sort dependencies out for the Linux builds. Perhaps we could do something similar for the Flatpak to keep it in sync, at least? I'll poke some folks. :)

3. Great! Thank you!

4. Personally I would be in favor of shipping Audacity _exactly how upstream intends_ on Flathub, because it means everyone gets the "official" experience. If there's contention around enabling networking, that's not really Flathub's concern imho, because it's your app. 🤷🏻 Flathub accepts proprietary apps and reviews the metainfo for accuracy, but for the most part, does not make judgements on how apps themselves should work. As long as it's accurately noted in the [OARS data](https://github.com/audacity/audacity/blob/bd85b51562427a230c35db5f6ad58a6234f134cb/help/audacity.appdata.xml#L27), I don't see the issue—and if people do have an issue, they're welcome to submit a fork to Flathub under a different name. I can chat a bit with the Flathub admin team about this issue specifically, though, because if that's something preventing the Flatpak from being better supported, we should remove that blocker.

Thank you again for your time! Even if we don't remove all the blockers today, this has been super enlightening and helpful. Hopefully we can systematically remove any remaining blockers, then maybe we can reassess the situation later down the line. :)

**Comment by dphaldes** [comment]
I think this should be reconsidered. Flatpak/Flathub are becoming the Linux app distribution medium. There are some distros that fully rely on it for application distribution.
> We put effort into making Audacity build without using the default dependency management, but we do not have the resources to test the result. For this reason, we leave Audacity packaging to the package maintainers. 

I would argue this is a reason in favour of flatpak. It is supposed to be the one solution that fits all. It solves dependency management for all distros in a common fashion and gives upstream(you) the control for application.

**Reply by Beryesa** [reply]
Are there any updates on the case with the 4.0 release, which seems more interested in providing binaries from the upstream?
